mlir/test/Dialect/Vector/vector-emulate-narrow-type-unaligned-dynamic.mlir

```diff
@@ -0,0 +1,53 @@
+// RUN: mlir-opt --test-emulate-narrow-int="arith-compute-bitwidth=1 memref-load-bitwidth=8" --cse --split-input-file %s | FileCheck %s
+
+// CHECK: #map = affine_map<()[s0, s1] -> ((s0 * 3 + s1) floordiv 4)>
+// CHECK: #map1 = affine_map<()[s0, s1] -> ((s0 * 3 + s1) mod 4)>
+func.func @vector_load_i2(%arg1: index, %arg2: index) -> vector<3x3xi2> {
+    %0 = memref.alloc() : memref<3x3xi2>
```

Comment:
Since these tests are for vector Ops, I suggest simplifying this test by passing `%0` as an input argument.

Reply:
Turns out it is an existing bug for emulating narrow types if the `%0` is an input argument. I will try to resolve that in another patch.

Reply:
I wouldn't worry about it. Basically to fix that we will have to change the function signature which is a whole other thing to do, or we will have to use `memref.reinterpret_casts` to flatten things, and we can do it if needed. Easiest is just use an alloc for now for lit tests

Reply:
OK, now I see what the issue is, thanks for explaining @MaheshRavishankar !

Just to explain where I'm coming from ... This a test for the Vector dialect. `memref.alloc` is a very complex and a powerful Op - its presence suggests that it's somehow significant for this test/transformation. But that's not the case, so it's just noise (and there's _a lot_ of noise in the Vector tests). Hence my request to remove it.

Since removing `memref.alloc` is not feasible ATM, could you add a TODO at the top of these tests? This way there's a note for our future-selves :)

Reply:
This style is probably written by me, or at least I reviewed the code. Making `%0` is a tricky in this kind of test because type converter is involved. I don't remember all the details. My impression is that there are patterns to replace the function argument types and it inserts a `unrealized_conversion_cast` is inserted. Because we want to flatten the memrefs and we have the emulation pattern for `memref.alloca`, so my easiest way was offloading it to `memref.alloca` which handles the case.

This style is clearer to me because that we do know that we want to flatten the memrefs. It is one of the core ideas in subtype emulation. Looping ABI changes (i.e., update the function argument type) seems to be a noise in this scope, IMO.

But yeah, more comments/documentations is better. @lialan can you help add a comment?